src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -496,6 +498,26 @@ private ImmutableArray<TExtension> GetAnalyzersForTypeNames(Assembly analyzerAss
 
                     Debug.Assert(type != null);
 
+                    // check if this references net framework, and issue a diagnostic if we don't allow that
+                    if (!_allowNetFramework)
+                    {
+                        const string ExpectedPublicKey = "B7-7A-5C-56-19-34-E0-89";
+                        foreach (var reference in analyzerAssembly.GetReferencedAssemblies())
+                        {
+                            if (reference.Name?.Equals("mscorlib", StringComparison.OrdinalIgnoreCase) == true)
+                            {
+                                var publicKeyToken = reference.GetPublicKeyToken();
+                                if (publicKeyToken is object && BitConverter.ToString(publicKeyToken).Equals(ExpectedPublicKey, StringComparison.OrdinalIgnoreCase))
```

Comment:
@jaredpar Not sure who else to ask. Does this seem like a sensible way to check for mscorlib references? (Do we have a utility that does it in the code already?)

Reply:
Rather than checking for mscorlib references I would look at the `TargetFrameworkAttribute` on the assembly. If the moniker begins with `.NETFRamework` when we should reject.

Reply:
Ok, and we're ok that you can create .NET Framework referencing assemblies without a `TargetFrameworkAttribute`? A default `csc.exe Generator.cs` invocation will target framework, but not add the attribute.

Reply:
Yes. Any check here is imprecise. The majority case is that they use our build infra,n not the raw compiler, which will add this attribute.